### PR TITLE
Remove warning in failed JSON RPC call, when still syncing relevant data

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Suppress warning logs when not yet synced resource is not there when calling eth_ module JSON RPC methods.

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [x] Other: logs

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs [hunk 1]
```diff
@@ -70,5 +70,33 @@ public enum SyncMode
     public static class SyncModeExtensions
     {
         public static bool NotSyncing(this SyncMode syncMode) => syncMode == SyncMode.WaitingForBlock || syncMode == SyncMode.Disconnected;
+
+        public static bool IsSyncingBodies(this SyncMode syncMode) =>
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs:74]: What about SyncMode.Disconnected? :)

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs:74]: Should we consider Disconnected as syncing?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs:74]: Above `NotSyncing` - says directly that we are not syncing :)
Lets keep the warning then.

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs [hunk 2]
```diff
@@ -69,6 +69,34 @@ public enum SyncMode
...
+            syncMode.HasFlag(SyncMode.FastHeaders) ||
+            syncMode.HasFlag(SyncMode.FastBodies) ||
+            syncMode.HasFlag(SyncMode.FastSync) ||
+            syncMode.HasFlag(SyncMode.StateNodes) ||
+            syncMode.HasFlag(SyncMode.SnapSync) ||
+            syncMode.HasFlag(SyncMode.BeaconHeaders) ||
+            syncMode.HasFlag(SyncMode.UpdatingPivot);
+
+        public static bool IsSyncingReceipts(this SyncMode syncMode) =>
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncMode.cs:83]: HadNotSyncReceipts? ReceiptNotSynced?

**Review by asdacap — APPROVED** [review]

**Pull request merged**